src-plugins/reformat/superResolutionProcess.cpp

```diff
@@ -95,61 +89,30 @@ void superResolutionProcess::setInput( medAbstractData *data, int channel )
 // Method to actually start the filter
 int superResolutionProcess::update ( void )
 {
-    if ( !d->input0 || !d->input1  || !d->input2  || !d->input3 )
+    if ( d->input0 && d->input1 && d->input2 && d->input3 )
     {
-        qDebug() << "in update method : d->input is NULL";
-        return -1;
+        // cast to unsigned char and set input in shapeBasedInterpolationFilter
+        bool res0 = castToUChar3(d->input0);
+        bool res1 = castToUChar3(d->input1);
+        bool res2 = castToUChar3(d->input2);
+        bool res3 = castToUChar3(d->input3);
+
+        if ( res0 == DTK_SUCCEED && res1 == DTK_SUCCEED && res2 == DTK_SUCCEED && res3 == DTK_SUCCEED )
+        {
+            connect(&d->shapeBasedInterpolationFilter , SIGNAL(progressed(int)), this, SIGNAL(progressed(int)));
+            d->shapeBasedInterpolationFilter.run();
```

Comment:
Isn't there a way that you set the inputs of the filter in this method? It's for readability.
Because now it's in the *cast()* method which thus do a bit more than casting.